lnwallet/channel.go

```diff
@@ -1893,6 +1893,33 @@ func (lc *LightningChannel) restorePendingRemoteUpdates(
 				payDesc.LogIndex)
 		}
 
+		var height uint64
+		var heightSet bool
+		if payDesc.EntryType != Add {
+			// Check whether this update is included on the pending commitment
+			// If so, we should set the removeCommitHeightRemote.
+			pendingRemoteCommitDiff, err := lc.channelState.RemoteCommitChainTip()
+			if err != nil && err != channeldb.ErrNoPendingCommit {
+				return err
+			}
+
+			if pendingRemoteCommitDiff != nil {
+				pendingCommitPoint := lc.channelState.RemoteNextRevocation
+				pendingRemoteCommit, err := lc.diskCommitToMemCommit(
```

Comment:
it looks like we will end up fetching the commitment multiple times for the same height?

Reply:
You're right this is slow, can just look it up once